ansible/roles/test/tasks/vlan_configure.yml

```diff
@@ -21,27 +22,25 @@
 - debug: var=vlan_ports_list
 - debug: var=vlan_intf_list
 
-- name: Flush all IP addresses on the LAGs
-  shell: ip addr flush {{ item.attachto }}
-  with_items:
-    - "{{ minigraph_portchannel_interfaces }}"
-  become: true
-
-- name: Delete all IP addresses on the LAGs in config DB
-  shell: docker exec -i database redis-cli -n 4 del "PORTCHANNEL_INTERFACE|{{ item.attachto }}|{{ (item.addr ~ '/' ~ item.mask)|ipaddr()|upper() }}"
+- name: Shutdown LAGs
+  shell: config interface shutdown {{ item.attachto }}
   with_items:
     - "{{ minigraph_portchannel_interfaces }}"
   become: true
 
-- name: Shutdown LAGs
-  shell: ifconfig {{ item.attachto }} down
+- name: Flush all IP addresses on the LAGs
+  shell: config interface ip remove {{ item.attachto }} {{ (item.addr ~ "/" ~ item.mask)|ipaddr()|upper }}
   with_items:
     - "{{ minigraph_portchannel_interfaces }}"
   become: true
 
+# wait some time for route, neighbor, next hop groups to be removed,
+# otherwise PortChannel RIFs are still referenced and won't be removed
+# cause below vlan_configuration.json fail to apply
 - name: sleep for some time
-  pause: seconds=10
+  pause: seconds=90
```

Comment:
Do we need 90 sec here? I think, 30sec would be reasonable.

Reply:
@prsunny Please refer to my email I sent to you. According to my test (on Mellanox SN2700) it takes up to 90 sec for all routes -> ECMP groups -> neighbors -> rifs to be removed from HW on master branch.
Max for 201811 is 30 sec.